### PR TITLE
[py] Replace devtools genrule with proper generate_devtools rule.

### PR DESCRIPTION
**Pull request by jameshilliard**
The existing genrule does not handle directories properly.

See [details](https://github.com/bazelbuild/rules_pkg/pull/647#issuecomment-1341254374).

### CODE REVIEW

File: py/private/generate_devtools.bzl [hunk 1]
```diff
@@ -0,0 +1,50 @@
+def _generate_devtools_impl(ctx):
+    outdir = ctx.actions.declare_directory("{}".format(ctx.attr.outdir))
```
Comment (aiuto) [inline py/private/generate_devtools.bzl]: Why the format()?  Just use the name.

File: py/generate.py [hunk 2]
```diff
@@ -988,7 +988,7 @@ def main(browser_protocol_path, js_protocol_path, output_path):
         browser_protocol_path,
         js_protocol_path,
     ]
-    output_path.mkdir(parents=True)
+    output_path.mkdir(parents=True, exist_ok=True)
```
Comment (aiuto) [inline py/generate.py]: I think you don't need the mkdir. Declare directory should have done it for you.

Comment (jameshilliard) [reply py/generate.py]: removed mkdir

File: py/private/generate_devtools.bzl (hunk 1 above)
Comment (jameshilliard) [reply py/private/generate_devtools.bzl]: removed format()

File: py/private/generate_devtools.bzl [hunk 3]
```diff
@@ -0,0 +1,50 @@
+def _generate_devtools_impl(ctx):
+    outdir = ctx.actions.declare_directory(ctx.attr.outdir)
+
+    args = ctx.actions.args()
+    args.add(ctx.attr.browser_protocol.files.to_list()[0])
```
Comment (shs96c) [inline py/private/generate_devtools.bzl]: `ctx.file.browser_protocol`

File: py/private/generate_devtools.bzl [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    args.add(ctx.attr.js_protocol.files.to_list()[0])
```
Comment (shs96c) [inline py/private/generate_devtools.bzl]: `ctx.file.js_protocol`

**Review by shs96c — COMMENTED** [review]
LGTM

**Comment by diemol** [comment]
@jameshilliard would you like to continue with this PR? Some comments were left.

**Comment by jameshilliard** [comment]
> @jameshilliard would you like to continue with this PR? Some comments were left.

Applied suggestions from comments and rebased.

**Comment by diemol** [comment]
Seems the build is now failing. Could you please check?

**Review by diemol — APPROVED** [review]
Thank you, @jameshilliard!

**Pull request merged**